docs/source/en/tutorials/custom_pipelines_components.md

```diff
@@ -0,0 +1,135 @@
+<!--Copyright 2023 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# Working with fully custom pipelines and components
+
+Diffusers supports the use [custom pipelines](../using-diffusers/contribute_pipeline) letting the users add any additional features on top of the [`DiffusionPipeline`]. However, it can get cumbersome if you're dealing with a custom pipeline where its components (such as the UNet, VAE, scheduler) are also custom. 
+
+We allow loading of such pipelines by exposing a `trust_remote_code` argument inside [`DiffusionPipeline`]. The advantage of `trust_remote_code` lies in its flexibility. You can have different levels of customizations for a pipeline. Following are a few examples:
+
+* Only UNet is custom 
+* UNet and VAE both are custom
+* Pipeline is custom 
+* UNet, VAE, scheduler, and pipeline are custom 
+
+With `trust_remote_code=True`, you can achieve perform of the above!
+
+This tutorial covers how to author your pipeline repository so that it becomes compatible with `trust_remote_code`. You'll use a custom UNet, a custom scheduler, and a custom pipeline for this purpose. 
+
+<Tip warning={true}>
+
+You should use `trust_remote_code=True` _only_ when you fully trust the code and have verified its usage.
+
+</Tip>
+
+## Pipeline components
+
+In the interest of brevity, you'll use the custom UNet, scheduler, and pipeline classes that we've already authored:
+
+```bash
+# Custom UNet
+wget https://huggingface.co/sayakpaul/custom_pipeline_remote_code/raw/main/unet/my_unet_model.py
+# Custom scheduler
+wget https://huggingface.co/sayakpaul/custom_pipeline_remote_code/raw/main/scheduler/my_scheduler.py
+# Custom pipeline
+wget https://huggingface.co/sayakpaul/custom_pipeline_remote_code/raw/main/my_pipeline.py
+```
+
+<Tip warning={true}>
+
+The above classes are just for references. We encourage you to experiment with these classes for desired customizations.
+
+</Tip>
```

Comment:
Ok to remove this warning now that we've already clarified that this is just an example.

```suggestion
```